### PR TITLE
Couple of fixes for i.MX 

### PR DESCRIPTION
**Pull request by dbaluta**
This adds hw_info field for sof_imx8_ops and then fixes DAI driver names to
match with the ones in topology. 

So, far this worked because internally we didn't use DAI driver names from imx/imx8.c, imx/imx8m.c files
but a custom generic DAI driver with DAIs defined in device trees.  

This approach, didn't receive a positive feedback from the community
so we have now found a solution to use directly DAIs defined inside the code.

### CODE REVIEW

**Comment by dbaluta** [comment]
Tested only separately.

File: sound/soc/sof/imx/imx8.c [hunk 1]
```diff
@@ -416,6 +416,13 @@ struct snd_sof_dsp_ops sof_imx8_ops = {
 	/* DAI drivers */
 	.drv = imx8_dai,
 	.num_drv = 1, /* we have only 1 ESAI interface on i.MX8 */
+
+	/* ALSA HW info flags */
+	.hw_info =	SNDRV_PCM_INFO_MMAP |
+			SNDRV_PCM_INFO_MMAP_VALID |
+			SNDRV_PCM_INFO_INTERLEAVED |
+			SNDRV_PCM_INFO_PAUSE |
+			SNDRV_PCM_INFO_NO_PERIOD_WAKEUP
```
Comment (plbossart) [inline sound/soc/sof/imx/imx8.c]: I already see this code in topic/sof-dev?

Comment (dbaluta) [reply sound/soc/sof/imx/imx8.c]: @plbossart, I thought the same thing *but* actually we have two dsp_ops in imx8.c. One for imx8x and one for imx8.
 
Same as you I could swear that this code was already in topic/sof-dev. Will triple check tommorrow.

File: sound/soc/sof/imx/imx8.c [hunk 2]
```diff
@@ -415,7 +426,7 @@ struct snd_sof_dsp_ops sof_imx8_ops = {
 
 	/* DAI drivers */
 	.drv = imx8_dai,
-	.num_drv = 1, /* we have only 1 ESAI interface on i.MX8 */
+	.num_drv = 2, /* for now we support 2 DAIs on i.MX8 */
```
Comment (ranj063) [inline sound/soc/sof/imx/imx8.c]: ARRAY_SIZE() maybe?

File: sound/soc/sof/imx/imx8.c (hunk 1 above)
Comment (plbossart) [reply sound/soc/sof/imx/imx8.c]: oh right, indeed there's one missing.

Comment (lyakh) [inline sound/soc/sof/imx/imx8.c]: I'd use a comma at the end of the statement - see how one after `.num_drv` above allowed you to just add one more without modifying the previous one too.

Comment (dbaluta) [reply sound/soc/sof/imx/imx8.c]: @lyakh ok, will fix.

File: sound/soc/sof/imx/imx8.c (hunk 2 above)
Comment (dbaluta) [reply sound/soc/sof/imx/imx8.c]: @ranj063 makes sense, will fix.

**Comment by dbaluta** [comment]
@plbossart @ranj063 @lyakh thanks for review. Addressed all the comments and resent.

**Pull request merged**